client/me/help/help-happiness-engineers/index.jsx

```diff
@@ -1,57 +1,49 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	PureRenderMixin = require( 'react-pure-render/mixin' );
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import React from 'react';
+import { shuffle } from 'lodash';
 
 /**
  * Internal dependencies
  */
-var HappinessEngineersStore = require( 'lib/happiness-engineers/store' ),
-	HappinessEngineersActions = require( 'lib/happiness-engineers/actions' ),
-	FormSectionHeading = require( 'components/forms/form-section-heading' ),
-	Gravatar = require( 'components/gravatar' );
-
-module.exports = React.createClass( {
-	displayName: 'HelpHappinessEngineers',
-
-	mixins: [ PureRenderMixin ],
-
-	componentDidMount: function() {
-		HappinessEngineersStore.on( 'change', this.refreshHappinessEngineers );
-		HappinessEngineersActions.fetch();
-	},
-
-	componentWillUnmount: function() {
-		HappinessEngineersStore.removeListener( 'change', this.refreshHappinessEngineers );
-	},
-
-	getInitialState: function() {
-		return {
-			happinessEngineers: HappinessEngineersStore.getHappinessEngineers()
-		};
-	},
-
-	refreshHappinessEngineers: function() {
-		this.setState( { happinessEngineers: HappinessEngineersStore.getHappinessEngineers() } );
-	},
-
-	render: function() {
-		if ( this.state.happinessEngineers.length === 0 ) {
-			return null;
-		}
-		return (
-			<div className="help-happiness-engineers">
-				{ this.translate( '{{headline}}We care about your happiness!{{/headline}}{{p}}They don\'t call us Happiness Engineers for nothing. If you need help, we\'re here for you!{{/p}}', {
+import FormSectionHeading from 'components/forms/form-section-heading';
+import Gravatar from 'components/gravatar';
+import QueryHappinessEngineers from 'components/data/query-happiness-engineers';
+import { getHappinessEngineers } from 'state/happiness-engineers/selectors';
+
+function HelpHappinessEngineers( props ) {
+	return (
+		<div className="help-happiness-engineers">
+			{ props.translate( '{{headline}}We care about your happiness!{{/headline}}' +
+				'{{p}}They don\'t call us Happiness Engineers for nothing. ' +
+				'If you need help, we\'re here for you!{{/p}}',
+				{
 					components: {
 						headline: <FormSectionHeading />,
 						p: <p className="help-happiness-engineers__description" />
 					}
-				} ) }
-				<div className="help-happiness-engineers__tray">
-					{ this.state.happinessEngineers.map( happinessEngineer => <Gravatar key={ happinessEngineer.avatar_URL } user={ { avatar_URL: happinessEngineer.avatar_URL } } size={ 42 } /> ) }
-				</div>
+				}
+				) }
+			<div className="help-happiness-engineers__tray">
+				{ shuffle( props.happinessEngineers ).map(
+					happinessEngineer => <Gravatar
+						key={ happinessEngineer }
+						user={ { avatar_URL: happinessEngineer } }
+						size={ 42 } /> )
+				}
 			</div>
-		);
+			{ props.happinessEngineers.length === 0 && <QueryHappinessEngineers /> }
```

Comment:
To expand on this, the reason for a selector, even when so minimal in logic as in my previous comment, is for maintainability's sake. While `happinessEngineers === null` makes sense to you now, can you guarantee it will in the future? Perhaps so in this simple case, but it's something to keep in mind. Alternatively, `! hasReceived && <QueryHappinessEngineers />` could make this intent clearer. Also, selectors "keeping state opaque" means that we could change the initial value and not have to update all components referencing it like this, since the return value of the selector will presumably stay consistent.

This is more a brain dump of my suggesting a selector, not something that needs to change here. It is a simple condition and I'm not too worried about it.